hetzner/storagebox.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from datetime import datetime
+
+__all__ = ['StorageBox', 'SubAccount', 'SubAccountManager']
+
+
+class SubAccount(object):
+    def __init__(self, conn, box_id_, result):
+        self.conn = conn
+        self.box_id_ = box_id_
+        self.update_info(result)
+
+    def update_info(self, result):
+        """
+        Update the information of the subaccount.
+        """
+        data = result['subaccount']
+
+        self.username = data['username']
+        self.accountid = data['accountid']
+        self.server = data['server']
+        self.homedirectory = data['homedirectory']
+        self.samba = data['samba']
+        self.ssh = data['ssh']
+        self.external_reachability = data['external_reachability']
+        self.webdav = data['webdav']
+        self.readonly = data['readonly']
+        self.createtime = datetime.strptime(data['createtime'], '%Y-%m-%d %H:%M:%S')
+        self.comment = data['comment']
+
+    def update(self, homedirectory, samba, ssh, external_reachability, webdav, readonly, comment):
+        result = self.conn.put('/storagebox/{0}/subaccount/{1}'.format(self.box_id_, self.username),
+                                {'homedirectory': homedirectory,
+                                 'samba': samba,
+                                 'ssh': ssh,
+                                 'external_reachability': external_reachability,
+                                 'webdav': webdav,
+                                 'readonly': readonly,
+                                 'comment': comment})
+
+        return result
```

Comment:
Hm, so far I've been trying to adhere to PEP-8, including limiting the line length, so for consistency I'd suggest:

```suggestion
        pathfmt = '/storagebox/{0}/subaccount/{1}'
        path = pathfmt.format(self.box_id_, self.username)
        data = {
            'homedirectory': homedirectory,
            'samba': samba,
            'ssh': ssh,
            'external_reachability': external_reachability,
            'webdav': webdav,
            'readonly': readonly,
            'comment': comment
        }
        return self.conn.put(path, data)
```

Another thing I'm considering as well is to drop support for Python 2 entirely and use type hints and `f`-strings, so if it doesn't bother you if this is included in the next major version rather than the next minor version you could even use `f'/storagebox/{self.box_id_}/subaccount/{self.username}'` instead.